projects/core/src/product/services/product-loading.service.spec.ts

```diff
@@ -62,103 +64,114 @@ describe('ProductLoadingService', () => {
+          )
         );
 
         const result: Product = await service
-          .get('testId', ['scope1', 'scope2'])
+          .get(code, ['scope1', 'scope2'])
           .pipe(take(1))
           .toPromise();
-        expect(result).toEqual({ name: 'test' });
+        expect(result).toEqual({ code, name: 'test' });
       });
 
       it('should take into account order of scopes', async () => {
-        let callNo = 0;
-        const productScopes = [
-          { name: 'first', code: 'a' },
-          { name: 'second', description: 'b' },
-        ];
-        spyOnProperty(ngrxStore, 'select').and.returnValue(() => () =>
-          new BehaviorSubject({
-            value: productScopes[callNo++], // serve different scope per call
-          })
+        store.dispatch(
+          new ProductActions.LoadProductSuccess(
+            { code, name: 'first', summary: 'a' },
+            'scope1'
+          )
+        );
+        store.dispatch(
+          new ProductActions.LoadProductSuccess(
+            { code, name: 'second', description: 'b' },
+            'scope2'
+          )
         );
+
         const result: Product = await service
-          .get('testId', ['scope1', 'scope2'])
+          .get(code, ['scope1', 'scope2'])
           .pipe(take(1))
           .toPromise();
-        expect(result).toEqual({ name: 'second', code: 'a', description: 'b' });
+        expect(result).toEqual({
+          code,
+          name: 'second',
+          summary: 'a',
+          description: 'b',
+        });
       });
 
       it('should take into account order of scopes for subsequent emissions', done => {
-        let callNo = 0;
-        const productSources = [
-          new BehaviorSubject<any>({
-            value: { name: 'first', code: 'a' },
-          }),
-          new BehaviorSubject<any>({
-            value: { name: 'second', description: 'b' },
-          }),
-        ];
-        spyOnProperty(ngrxStore, 'select').and.returnValue(() => () =>
-          productSources[callNo++]
+        const action1scope1 = new ProductActions.LoadProductSuccess(
+          { code, name: 'first', summary: 'a' },
+          'scope1'
+        );
+        const action1scope2 = new ProductActions.LoadProductSuccess(
+          { code, name: 'second', description: 'b' },
+          'scope2'
+        );
+        const action2scope1 = new ProductActions.LoadProductSuccess(
+          { code, name: 'third', summary: 'c' },
+          'scope1'
+        );
+        const action2scope2 = new ProductActions.LoadProductSuccess(
+          { code, name: 'fourth', description: 'e' },
+          'scope2'
         );
 
         const results: Product[] = [];
         service
-          .get('testId', ['scope1', 'scope2'])
+          .get(code, ['scope1', 'scope2'])
           .pipe(take(3))
           .subscribe({
             next: res => {
               results.push(res);
             },
             complete: () => {
               expect(results).toEqual([
```

Comment:
That's a nice pattern for multiple emissions tests :)

Reply:
TBH I should use `scan` operator here :-) but idea would be the same.